types/argon2-browser/index.d.ts

```diff
@@ -1,13 +1,14 @@
-// Type definitions for argon2-browser 1.1
+// Type definitions for argon2-browser 1.5
 // Project: https://github.com/antelle/argon2-browser#readme
 // Definitions by: Ivan Gabriele <https://github.com/ivangabriele>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.1
 
-export namespace argon2 {
-    function hash(options: Argon2BrowserHashOptions): Promise<Argon2BrowserHashResult>;
+declare namespace argon2 {
```

Comment:
I think the right thing to do here is to keep this as a module (for AMD) and declare it as a UMD global with

```ts
export as namespace argon2;
```

Reply:
I'm not confident enough with TS modules to have a strong opinion on this. However, for the very same reason, I'm not sure what you expect from me here.

The problem I have is that only UMD global seem to work for this library. I had no success with any other import. If I do
```ts
export namespace argon2 { … }
export as namespace argon2;
```

I get errors like this:
```
ERROR: 28:9   expect  TypeScript@next compile error: 
'argon2' refers to a UMD global, but the current file is a module. Consider adding an import instead.
```

Thanks!

Reply:
I think the main issue here is that the shape of the module in AMD and the shape of the module in Commonjs (i.e. node.js) are different. In the former, it's an object with three properties (hash, verify, and ArgonType), and in the latter, it's an object with one property (argon2), which is the object with three properties.

There's a way to have the module have one shape (i.e. the current, commonjs shape) when imported, but the other shape when used globally; it's kind of tricky though. It might be better to either fix the underlying library (if you can) to have a more sensical shape in commonjs, or to just commit to this declaration file only reflecting the AMD and global versions of the module and not the commonjs version. What do you want to do?

Reply:
@uniqueiniquity thanks. I guess it would be better to contribute to the original project to make their bundle more viable. Yet I'm unsure I'll be able to make it anytime soon since I'd have to try my hand on JS packaging and it looks kind of messy. I'll keep my type stubs hardcoded into my project at the moment.